Documentation/Diagnostics/PH2117.md

```diff
@@ -0,0 +1,53 @@
+# PH2117: Avoid Unnecessary Where()
+
+| Property | Value  |
+|--|--|
+| Package | [Philips.CodeAnalysis.MaintainabilityAnalyzers](https://www.nuget.org/packages/Philips.CodeAnalysis.MaintainabilityAnalyzers) |
+| Diagnostic ID | PH2117 |
+| Category  | [Maintainability](../Maintainability.md) |
+| Analyzer | [AvoidUnnecessaryWhereAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/master/Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidUnnecessaryWhereAnalyzer.cs)
+| CodeFix  | No |
+| Severity | Error |
+| Enabled By Default | Yes |
+
+## Introduction
+
+For LINQ statements with both a `Where()` and a `Any()`, `Count()`, `First()`, `Last()`, or `Single()` method, can be written more compact by using the overload with a predicate.
```

Comment:
LINQ statements with both a `Where()` and one of `Any()`, `Count()`, `First()`, `Last()`, or `Single()` method can be written more compactly by using the overload with a predicate.

(Remove "For", consider adding "one of", remove comma, and add 'ly' to compact.)

Reply:
All done